--- pump-0.8.11/pump.c.nobootp	Thu Mar  1 02:15:22 2001
+++ pump-0.8.11/pump.c	Thu Jul 19 18:29:55 2001
@@ -758,7 +758,7 @@
     int killDaemon = 0;
     int winId = 0;
     int release = 0, renew = 0, status = 0, lookupHostname = 0, nodns = 0;
-    int nogateway = 0;
+    int nogateway = 0, nobootp = 0;
     struct command cmd, response;
     char * configFile = "/etc/pump.conf";
     struct pumpOverrideInfo * overrides;
@@ -788,6 +788,8 @@
 			N_("Don't update resolv.conf"), NULL },
 	    { "no-gateway", '\0', POPT_ARG_NONE, &nogateway, 0,
 			N_("Don't set a gateway for this interface"), NULL },
+	    { "no-bootp", '\0', POPT_ARG_NONE, &nobootp, 0,
+	                N_("Ignore non-DHCP BOOTP responses"), NULL },
 	    { "win-client-ident", '\0', POPT_ARG_NONE, &winId, 0,
 			N_("Set the client identifier to match Window's") },
 	    /*{ "test", 't', POPT_ARG_NONE, &test, 0,
@@ -831,6 +833,8 @@
 	flags |= PUMP_FLAG_FORCEHNLOOKUP;
     if (nodns)
 	overrides->flags |= OVERRIDE_FLAG_NODNS;
+    if (nobootp)
+	overrides->flags |= OVERRIDE_FLAG_NOBOOTP;
     if (nogateway)
 	overrides->flags |= OVERRIDE_FLAG_NOGATEWAY;
 
--- pump-0.8.11/dhcp.c.nobootp	Tue Feb 27 12:03:02 2001
+++ pump-0.8.11/dhcp.c	Thu Jul 19 18:29:55 2001
@@ -1321,8 +1321,11 @@
 
     syslog (LOG_DEBUG, "PUMP: sending discover\n");
 
+    if (override && (override->flags & OVERRIDE_FLAG_NOBOOTP))
+      syslog (LOG_DEBUG, "PUMP: Ignoring non-DHCP BOOTP responses\n");
+ 
     if ((chptr = handleTransaction(s, override, &breq, &bresp, &broadcastAddr,
-				   NULL, 1, startTime, DHCP_TYPE_OFFER))) {
+				   NULL, (override && (override->flags & OVERRIDE_FLAG_NOBOOTP))?0:1, startTime, DHCP_TYPE_OFFER))) {
 	close(s);
 	pumpDisableInterface(intf->device);
 	return chptr;
--- pump-0.8.11/pump.h.nobootp	Thu Feb  1 17:58:16 2001
+++ pump-0.8.11/pump.h	Thu Jul 19 18:29:55 2001
@@ -74,6 +74,7 @@
 #define OVERRIDE_FLAG_NODNS		(1 << 0)
 #define OVERRIDE_FLAG_NONISDOMAIN	(1 << 1)
 #define OVERRIDE_FLAG_NOGATEWAY		(1 << 2)
+#define OVERRIDE_FLAG_NOBOOTP           (1 << 3)
 
 struct pumpOverrideInfo {
     struct pumpNetIntf intf;
--- pump-0.8.11/config.c.nobootp	Thu Jul 19 18:30:02 2001
+++ pump-0.8.11/config.c	Thu Jul 19 18:30:20 2001
@@ -182,6 +182,12 @@
 		return 1;
 	    }
 	    override->flags |= OVERRIDE_FLAG_NODNS;
+	} else if (!strcmp(start, "nobootp")) {
+	    if (*rest) {
+		parseError(*lineNum, "unexpected argument to nobootp directive");
+		return 1;
+	    }
+	    override->flags |= OVERRIDE_FLAG_NOBOOTP;
 	} else if (!strcmp(start, "nogateway")) {
 	    if (*rest) {
 		parseError(*lineNum, "unexpected argument to nogateway directive");
