--- pump-0.8.11.orig/Makefile	Thu Mar  1 08:15:22 2001
+++ pump-0.8.11/Makefile	Wed Feb 27 09:01:52 2002
@@ -1,18 +1,17 @@
 SBINPATH = $(RPM_BUILD_ROOT)/sbin
-USRSBINPATH = $(sbindir)
-USRLIBPATH = $(libdir)
-INCPATH = $(includedir)
-MAN8PATH = $(mandir)/man8
-CFLAGS = -I. -Wall -g $(RPM_OPT_FLAGS) -D__STANDALONE__ -DVERSION=\"$(VERSION)\"
-CVSROOT = $(shell cat CVS/Root)
+USRSBINPATH = $(RPM_BUILD_ROOT)/usr/sbin
+USRLIBPATH = $(RPM_BUILD_ROOT)/usr/lib
+INCPATH = $(RPM_BUILD_ROOT)/usr/include
+MAN8PATH = $(RPM_BUILD_ROOT)/usr/man/man8
+CFLAGS = -Wall $(COPT_FLAGS) -D__STANDALONE__ -DVERSION=\"$(VERSION)\"
 
-LOADLIBES = /usr/lib/libpopt.a -lresolv
+LOADLIBES = -lpopt -lresolv
 LDFLAGS = -g
 
 VERSION=0.8.11
 CVSTAG = r$(subst .,-,$(VERSION))
 
-all:	pump netconfig
+all:	pump
 
 pump:	pump.o config.o libpump.a(dhcp.o)
 
@@ -34,7 +33,6 @@
 install:
 	mkdir -p $(SBINPATH) $(MAN8PATH) $(USRSBINPATH)
 	mkdir -p $(USRLIBPATH) $(INCPATH)
-	install -m 755 -s netconfig $(USRSBINPATH)/netconfig
 	install -m 755 -s pump $(SBINPATH)/pump
 	install -m 644 pump.8 $(MAN8PATH)
 	install -m 644 libpump.a $(USRLIBPATH)
