--- pump-0.8.11/pump.c.sopwith	Sun Aug 26 18:19:18 2001
+++ pump-0.8.11/pump.c	Sun Aug 26 18:21:42 2001
@@ -67,7 +67,7 @@
 	struct {
 	    char device[20];
 	    int flags;
-	    int reqLease;			/* in hours */
+	    int reqLease;			/* in seconds */
 	    char reqHostname[200];
 	} start;
 	int result;				/* 0 for success */
@@ -803,7 +803,8 @@
     int rc;
     int test = 0;
     int flags = 0;
-    int lease = 12;
+    int lease_hrs = 0;
+    int lease = 12*3600;
     int killDaemon = 0;
     int winId = 0;
     int release = 0, renew = 0, status = 0, lookupHostname = 0, nodns = 0;
@@ -823,8 +824,10 @@
 			N_("iface") },
 	    { "kill", 'k', POPT_ARG_NONE, &killDaemon, 0,
 			N_("Kill daemon (and disable all interfaces)"), NULL },
-	    { "lease", 'l', POPT_ARG_INT, &lease, 0,
+	    { "lease", 'l', POPT_ARG_INT, &lease_hrs, 0,
 			N_("Lease time to request (in hours)"), N_("hours") },
+	    { "leasesecs", 'L', POPT_ARG_INT, &lease, 0,
+			N_("Lease time to request (in seconds)"), N_("seconds") },
 	    { "lookup-hostname", '\0', POPT_ARG_NONE, &lookupHostname, 0,
 			N_("Force lookup of hostname") },
 	    { "release", 'r', POPT_ARG_NONE, &release, 0,
@@ -906,7 +909,10 @@
 	cmd.type = CMD_STARTIFACE;
 	strcpy(cmd.u.start.device, device);
 	cmd.u.start.flags = flags;
-	cmd.u.start.reqLease = lease * 60 * 60;
+	if(lease_hrs)
+		cmd.u.start.reqLease = lease_hrs * 60 * 60;
+	else
+		cmd.u.start.reqLease = lease;
 	strcpy(cmd.u.start.reqHostname, hostname);
     }
 
